### PR TITLE
[MLIR][ROCDL] Added `SchedGroupBarrier` and `IglpOpt` ops

### PR DESCRIPTION
**Pull request by ravil-mobile**
This PR adds missing `sched.group.barrier` and `rocdl.iglp.opt` ops to the ROCDL dialect (see [here](https://github.com/llvm/llvm-project/blob/ec78f0da0e9b1b8e2b2323e434ea742e272dd913/clang/include/clang/Basic/BuiltinsAMDGPU.def#L66-L68)). The ops are converted to the corresponding intrinsic calls during the translation from MLIR to LLVM IRs. This intrinsics are hints to the instruction scheduler of the AMDGPU backend.

### CODE REVIEW

**Comment by giuseros** [comment]
Hi @ravil-mobile , could you add the minimal tests as done for the other intrinsics?

**Comment by giuseros** [comment]
Also, could you add a more descriptive message to this PR?

**Comment by manupak** [comment]
cc: @arsenm

**Review by arsenm — COMMENTED** [review]
Missing tests

**Comment by ravil-mobile** [comment]
Hi @giuseros , @arsenm @manupak ,

I've just added the tests. Can you have a look, please?

**Review by giuseros — CHANGES_REQUESTED** [review]

File: mlir/test/Dialect/LLVMIR/rocdl.mlir
Comment (ravil-mobile) [reply mlir/test/Dialect/LLVMIR/rocdl.mlir]: Thanks @giuseros, I guess it is done. Can you have a look?

Comment (giuseros) [reply mlir/test/Dialect/LLVMIR/rocdl.mlir]: Yep, looks perfect now!

**Review by giuseros — APPROVED** [review]

**Comment by giuseros** [comment]
Let's wait for @arsenm's approval to merge

**Review by arsenm — APPROVED** [review]

**Pull request merged**
